main/src/main/java/org/mobilitydata/gtfsvalidator/validator/UrlConsistencyValidator.java

```diff
@@ -0,0 +1,244 @@
+ * </ul>
+ *
+ * <p>Generated notice:
+ *
+ * <ul>
+ *   <li>{@link SameRouteAndAgencyUrlNotice} - for a record from "routes.txt", the value of
+ *   {@code routes.route_url} is shared by a record from "agency.txt".
+ *   <li>{@link SameStopAndAgencyUrlNotice} - for a record from "stops.txt", the value of
+ *   {@code stops.route_url} is shared by a record from "agency.txt".
+ *   *   <li>{@link SameStopAndRouteUrlNotice} - for a record from "stops.txt", the value of
+ *   {@code stops.route_url} is shared by a record from "routes.txt".
+ * </ul>
+ */
+@GtfsValidator
+public class UrlConsistencyValidator extends FileValidator {
+
+  private final GtfsAgencyTableContainer agencyTable;
+  private final GtfsRouteTableContainer routeTable;
+  private final GtfsStopTableContainer stopTable;
+
+  @Inject
+  UrlConsistencyValidator(GtfsAgencyTableContainer agencyTable,
+      GtfsRouteTableContainer routeTable, GtfsStopTableContainer stopTable) {
+    this.agencyTable = agencyTable;
+    this.routeTable = routeTable;
+    this.stopTable = stopTable;
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    Map<String, GtfsAgency> agencyByUrlMap = agenciesByUrlMap(agencyTable);
+    for (GtfsRoute route : routeTable.getEntities()) {
+      if (!route.hasRouteUrl()) {
+        return;
+      }
+      GtfsAgency agency = agencyByUrlMap.get(route.routeUrl().toLowerCase());
+      if (agency != null) {
+        noticeContainer.addValidationNotice(
+            new SameRouteAndAgencyUrlNotice(
+                route.csvRowNumber(),
+                route.routeId(),
+                agency.agencyName(),
+                route.routeUrl(),
+                agency.csvRowNumber()
+            ));
+      }
+    }
+    for (GtfsStop stop : stopTable.getEntities()) {
+      if (!stop.hasStopUrl()) {
+        return;
+      }
+      GtfsAgency agency = agencyByUrlMap.get(stop.stopUrl().toLowerCase());
+      if (agency != null) {
+        noticeContainer.addValidationNotice(
+            new SameStopAndAgencyUrlNotice(
+                stop.csvRowNumber(),
+                stop.stopId(),
+                agency.agencyName(),
+                stop.stopUrl(),
+                agency.csvRowNumber()
+            ));
+      }
+    }
+    Map<String, GtfsRoute> routesByUrlMap = routesByUrlMap(routeTable);
+    Map<String, GtfsStop> stopsByUrlMap = stopsByUrlMap(stopTable);
+
+    Maps.filterEntries(stopsByUrlMap,
+        entry -> routesByUrlMap.get(entry.getValue().stopUrl()) != null)
+        .values().forEach(stopWithDuplicateUrl -> {
+      noticeContainer.addValidationNotice(
+          new SameStopAndRouteUrlNotice(
+              stopWithDuplicateUrl.csvRowNumber(),
+              stopWithDuplicateUrl.stopId(),
+              stopWithDuplicateUrl.stopUrl(),
+              routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).routeId(),
+              routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).csvRowNumber()
+          ));
+    });
```

Comment:
This code looks very cool but I see a couple of issues with it.

1. (big) If multiple stops have the same `stop_url`, then a notice is generated only for one of them.
2. (tiny) Double lookup in routesByUrlMap.

I suggest to rewrite this block and move its logic into the cycle above:

```
for (GtfsStop stop : stopTable.getEntities()) {
   // Examine both agency and route urls here.
```

Reply:
Your suggestion makes the code more straightforward and easy to read, thanks for flagging this! You can review the changes in #1003.